stl/inc/format

```diff
@@ -2557,8 +2557,202 @@ template <_Format_supported_charT _CharT, class _Traits>
+            _Grow(_New_capacity);
+        }
+    }
+
+    void push_back(const _Ty _Value) {
+        try_reserve(_Size + 1);
+        _Ptr[_Size++] = _Value;
+    }
+};
+
+struct _Fmt_buffer_traits {
+    explicit _Fmt_buffer_traits(ptrdiff_t) {}
+    _NODISCARD size_t count() const noexcept {
+        return 0;
+    }
+    _NODISCARD size_t limit(const size_t _Size) noexcept {
+        return _Size;
+    }
+};
+
+class _Fmt_fixed_buffer_traits {
+private:
+    ptrdiff_t _Count = 0;
+    ptrdiff_t _Limit;
+
+public:
+    explicit _Fmt_fixed_buffer_traits(const ptrdiff_t _Lim) noexcept : _Limit(_Lim) {}
+    _NODISCARD size_t count() const noexcept {
+        return static_cast<size_t>(_Count);
+    }
+    _NODISCARD size_t limit(const size_t _Size) noexcept {
+        size_t _Avail = static_cast<size_t>(_Limit > _Count ? _Limit - _Count : 0);
+        _Count += _Size;
+        return _Size < _Avail ? _Size : _Avail;
+    }
+};
+
+inline constexpr size_t _Fmt_buffer_size = 256;
+
+template <class _OutputIt, class _Ty, class _Traits = _Fmt_buffer_traits>
+class _Fmt_iterator_buffer final : public _Traits, public _Fmt_buffer<_Ty> {
+private:
+    _OutputIt _Out;
+    _Ty _Data[_Fmt_buffer_size];
+
+protected:
+    void _Grow(size_t) final override {
+        if (this->size() == _Fmt_buffer_size) {
+            _Flush();
+        }
+    }
+    void _Flush() {
+        auto _Size = this->size();
+        this->clear();
+        _Out = _STD _Copy_unchecked(_Data, _Data + this->limit(_Size), _STD move(_Out));
+    }
+
+public:
+    explicit _Fmt_iterator_buffer(_OutputIt _Output, ptrdiff_t _Size = _Fmt_buffer_size)
+        : _Traits(_Size), _Fmt_buffer<_Ty>(_Data, 0, _Fmt_buffer_size), _Out(_STD move(_Output)) {}
+    ~_Fmt_iterator_buffer() {
+        if (this->size() != 0) {
+            _Flush();
+        }
+    }
+
+    _NODISCARD _OutputIt out() {
+        _Flush();
+        return _STD move(_Out);
+    }
+    _NODISCARD ptrdiff_t count() const noexcept {
+        return static_cast<ptrdiff_t>(_Traits::count() + this->size());
+    }
+};
+
+template <class _Ty>
+class _Fmt_iterator_buffer<_Ty*, _Ty> final : public _Fmt_buffer<_Ty> {
+protected:
+    void _Grow(size_t) final override {}
+
+public:
+    explicit _Fmt_iterator_buffer(_Ty* _Out, ptrdiff_t = 0) : _Fmt_buffer<_Ty>(_Out, 0, ~size_t()) {}
+
+    _NODISCARD _Ty* out() noexcept {
+        return this->end();
+    }
+};
+
+template <class _Container>
+class _Fmt_iterator_buffer<_STD back_insert_iterator<_Container>,
+    enable_if_t<_STD contiguous_iterator<typename _Container::iterator>, typename _Container::value_type>>
+    final : public _Fmt_buffer<typename _Container::value_type> {
+private:
+    _Container& _Cont;
+
+    struct _Accessor : _STD back_insert_iterator<_Container> {
+        _Accessor(_STD back_insert_iterator<_Container> _Iter) : back_insert_iterator<_Container>(_Iter) {}
+        using _STD back_insert_iterator<_Container>::container;
+    };

```

Comment:
What's the point of this `_Accessor` struct? Can't it just be `back_insert_iterator<_Container>`?

Reply:
To access the container member.

Reply:
Ok, makes sense...